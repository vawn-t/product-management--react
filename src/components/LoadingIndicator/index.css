@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

.loading-indicator {
  position: absolute;
  top: 50%;
  right: 50%;
  width: 50px;
  height: 50px;
  border: 10px solid #f3f3f3;
  /* Light grey */
  border-top: 10px solid #383636;
  /* Black */
  border-radius: 50%;
  animation: spinner 1.5s linear infinite;
}

.spinner-container {
  position: fixed;
  width: 100%;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0.5);
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 9999;
}
