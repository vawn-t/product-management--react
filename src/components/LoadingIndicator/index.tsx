// Libraries
import React, { memo } from 'react';

// Style
import './index.css';

const LoadingIndicator: React.FC = () => {
  return (
    <div className='spinner-container'>
      <div className='loading-indicator' />
    </div>
  );
};

export default memo(LoadingIndicator);
