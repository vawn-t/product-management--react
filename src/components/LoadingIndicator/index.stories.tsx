// Libraries
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Components
import LoadingIndicator from './index';

// Styles
import '@assets/styles/reset.css';
import '@assets/styles/App.css';
import '@assets/styles/variables.css';

export default {
  title: 'Components/LoadingIndicator',
  component: LoadingIndicator
} as ComponentMeta<typeof LoadingIndicator>;

const Template: ComponentStory<typeof LoadingIndicator> = (args) => (
  <LoadingIndicator {...args} />
);

export const Default = Template.bind({});
Default.args = {};
