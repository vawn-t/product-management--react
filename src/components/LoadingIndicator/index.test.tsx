// Libraries
import React from 'react';
import { create } from 'react-test-renderer';

// Components
import LoadingIndicator from './index';

describe('button component', () => {
  test('should render correctly', () => {
    const defaultTree = create(<LoadingIndicator />).toJSON();
    expect(defaultTree).toMatchSnapshot();
  });
});
