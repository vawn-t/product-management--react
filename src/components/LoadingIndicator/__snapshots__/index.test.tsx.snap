// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should render correctly 1`] = `
<div
  className="spinner-container"
>
  <div
    className="loading-indicator"
  />
</div>
`;
